chatgpt/nodes/validator.py

```diff
@@ -67,18 +131,119 @@ def __init__(self, config=None):
             self.lock = asyncio.Lock()
 
             vana.logging.info(
-                f"Running validator {self.node_server} on network: {self.config.chain.chain_endpoint} with dlpuid: {self.config.dlpuid}"
+                f"Running validator on network: {self.config.chain.chain_endpoint} with dlpuid: {self.config.dlpuid}"
             )
 
-    async def validate(self,
-                       message: chatgpt.protocol.ValidationMessage) -> chatgpt.protocol.ValidationMessage:
-        """
-        Proof of Contribution: Processes a validation request
-        :param message: The validation message
-        :return: The validation message with the output fields filled
-        """
-        vana.logging.info(f"Received {message.input_url} and encrypted key: {message.input_encryption_key}")
-        return await proof_of_contribution(message)
+            if not hasattr(self.state, 'needs_peer_scoring'):
+                setattr(self.state, 'needs_peer_scoring', [])
+
+    def record_file_score(self, file_id: int, score_data: Dict[str, Any]):
+        active_validators = self.get_active_validators()
+        current_block = self.chain_manager.get_current_block()
+
+        task = PeerScoringTask(
+            file_id=file_id,
+            active_validators=active_validators,
+            own_submission=score_data,
+            added_at_block=current_block
+        )
+
+        self.state.needs_peer_scoring.append(task)
+        self.state.save()
+
+    def get_active_validators(self) -> List[str]:
```

Comment:
It may not be needed right away, but this file is getting a bit cognitively heavy. I wonder if we can move some functions to utils, like get_active_validators and the peer scoring stuff.